src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,959 @@
+                    return true
+                }
+            }
+            return false
+        } else if (property === Property.Prime) {
+            return isPrime(num)
+        } else if (property === Property.Negative) {
+            return num < 0n
+        } else if (property === Property.Even) {
+            return num % 2n === 0n
+        } else if (property === Property.Divisible_By_Three) {
+            return num % 3n === 0n
+        } else if (property === Property.Divisible_By_Four) {
+            return num % 4n === 0n
+        } else if (property === Property.Divisible_By_Five) {
+            return num % 5n === 0n
+        } else if (property === Property.Divisible_By_Six) {
+            return num % 6n === 0n
+        } else if (property === Property.Divisible_By_Seven) {
+            return num % 7n === 0n
+        } else if (property === Property.Divisible_By_Eight) {
+            return num % 8n === 0n
+        } else if (property === Property.Ends_With_One) {
+            return num % 10n === 1n
+        } else if (property === Property.Ends_With_Two) {
+            return num % 10n === 2n
+        } else if (property === Property.Ends_With_Three) {
+            return num % 10n === 3n
+        } else if (property === Property.Ends_With_Four) {
+            return num % 10n === 4n
+        } else if (property === Property.Ends_With_Five) {
+            return num % 10n === 5n
+        } else if (property === Property.Ends_With_Six) {
+            return num % 10n === 6n
+        } else if (property === Property.Ends_With_Seven) {
+            return num % 10n === 7n
+        } else if (property === Property.Ends_With_Eight) {
+            return num % 10n === 8n
+        } else if (property === Property.Ends_With_Nine) {
+            return num % 10n === 9n
+        } else if (property === Property.Ends_With_Zero) {
+            return num % 10n === 0n
+        } else if (property === Property.Sum_Of_Two_Squares) {
+            return isSumOfTwoSquares(num)
+        } else if (property === Property.Triangular_Number) {
+            return isPolygonal(num, 3n)
+        } else if (property === Property.Square_Number) {
+            return isPolygonal(num, 4n)
+        } else if (property === Property.Pentagonal_Number) {
+            return isPolygonal(num, 5n)
+        } else if (property === Property.Hexagonal_Number) {
+            return isPolygonal(num, 6n)
+        } else if (property === Property.Heptagonal_Number) {
+            return isPolygonal(num, 7n)
+        } else if (property === Property.Octagonal_Number) {
+            return isPolygonal(num, 8n)
+        } else if (property === Property.Abundant) {
+            return isAbundant(num)
+        } else if (property === Property.Perfect) {
+            return isPerfect(num)
+        } else if (property === Property.Deficient) {
+            return !isPerfect(num) && !isAbundant(num)
+        } else if (property === Property.Semi_Prime) {
```

Comment:
The problem with this is that currently the Property ```enum``` is used to populate the drop-down menu automatically.  So I'm not sure what effect changing this would have on the drop-down menu... advice?

Reply:
The key here is to replace a huge if/then/else sequence with a single call to a function selected from a dictionary of implementations of the different properties.

Reply:
(This is not related to the drop-down in the dialog, but rather to the cleanliness of the code implementing the properties.)

Reply:
No, I think it is related?  The ```Property``` enum is automatically used to populate the drop-down menu.  So if I change it into something else, where will the drop-down menu populate from?  (I understand the code cleanliness issue, but this ```Property``` enum is playing two roles, so if I change it into functions, I'm not sure what happens to the drop down.)

Reply:
The enum should still exist. There should just also be an object whose keys are Property enum instances and whose values are the corresponding functions for computing the property. Then that whole if/then block becomes something like `propertyIndicatorFunction[property](num)`. Nice, eh?

Reply:
Ok great, that's the advice I was looking for.

Reply:
Done; please take a look.  The only issue is that there's an unused variable lint rule I sidestepped because I don't know how to solve it.  Advice on resolving that would be appreciated.

Reply:
It looks like this was done in https://github.com/numberscope/frontscope/pull/230/commits/f23f1284c6ce2320bbc9e669032e5ca11f7ac73e and https://github.com/numberscope/frontscope/pull/230/commits/4fe5d6c2065ca757cff1b71534e0facacad4164a.

Reply:
Looks like it went from 63 lines of code to 35! That's great. I did a bit of manual regression testing, and from what I can tell this is working as intended. Resolving.

Reply:
Wait, I want to take a look at that unused variable issue. Un-resolving...

Reply:
Ah, it looks like @katestange fixed the lint error in https://github.com/numberscope/frontscope/pull/230/commits/fa6bff641013846306dd508cc2ecb76f30e920d1.